syntax = "proto3";

package grpc;

message Empty {}

message Note {
    string name = 1;
    string message = 2;
    int64 timestamp = 3;
}

message TimeRequest {
    string client_id = 1;
    int64 client_time = 2;
}

message TimeResponse {
    int64 server_time = 1;
}

service ChatServer {
    rpc ChatStream (Empty) returns (stream Note);
    rpc SendNote (Note) returns (Empty);
    rpc SyncTime (TimeRequest) returns (TimeResponse);
}
